### PR TITLE
Prepare 'next' branch for ESR-91

### PR DESCRIPTION
**Pull request by moztcampbell**
Apply fixes to the next branch before creating the esr91 branch itself. This is a bit of a manual process, but at the moment it is once a year so I'm not sure we need to streamline it.

### CODE REVIEW

**Comment by moztcampbell** [comment]
I'm working though the list from @ewlsh and updating Migration Guide too

File: meson.build [hunk 1]
```diff
@@ -1,13 +1,13 @@
-project('spidermonkey-embedding-examples', 'cpp', version: 'esr90',
+project('spidermonkey-embedding-examples', 'cpp', version: 'esr91',
     meson_version: '>= 0.43.0',
-    default_options: ['cpp_std=c++17', 'warning_level=3'])
+    default_options: ['cpp_std=c++17', 'warning_level=1'])
 
 cxx = meson.get_compiler('cpp')
 
-args = ['-std=c++17', '-Wall']
+args = []
```
Comment (ptomato) [inline meson.build:7]: We will still need `-std=c++17`, no?

File: meson.build [hunk 2]
```diff
@@ -1,13 +1,13 @@
-project('spidermonkey-embedding-examples', 'cpp', version: 'esr90',
+project('spidermonkey-embedding-examples', 'cpp', version: 'esr91',
     meson_version: '>= 0.43.0',
-    default_options: ['cpp_std=c++17', 'warning_level=3'])
+    default_options: ['cpp_std=c++17', 'warning_level=1'])
```
Comment (ptomato) [inline meson.build]: Does esr91 cause a bunch of new warnings? Maybe we should add exceptions lower down in `test_warning_args`?

**Review by ptomato — APPROVED** [review]

File: meson.build (hunk 1 above)
Comment (moztcampbell) [reply meson.build:7]: That is covered by the `cpp_std=c++17` above in the default_options (which is what meson was complaining about so I removed the line from args).

File: meson.build (hunk 2 above)
Comment (moztcampbell) [reply meson.build]: Oh, that wasn't supposed to stay in

File: meson.build (hunk 1 above)
Comment (ptomato) [reply meson.build:7]: Oh, right, I missed that.

**Review by ptomato — APPROVED** [review]
Looks good!

On a side note, it's really nice that all of the entries in the migration guide are simple moves or renames or other 'mechanical' refactors!

**Pull request merged**